modelling/src/main/java/org/axonframework/modelling/command/Repository.java

```diff
@@ -59,4 +59,18 @@
      * @throws Exception when the factoryMethod throws an exception
      */
     Aggregate<T> newInstance(Callable<T> factoryMethod) throws Exception;
+
+    /**
+     * Loads an aggregate from the repository. If the aggregate is not found it creates the aggregate using the
+     * specified
+     * {@code factoryMethod}.
+     *
+     * @param aggregateIdentifier The identifier of the aggregate to load
+     * @param expectedVersion     The expected version of the loaded aggregate
+     * @param factoryMethod       The method to create the aggregate's root instance
+     * @return The aggregate root with the given identifier.
+     */
+    default Aggregate<T> loadOrCreate(String aggregateIdentifier, Long expectedVersion, Callable<T> factoryMethod) {
```

Comment:
Shouldn't we expect a thrown `Exception` from the `factoryMethod`, like the `newInstance(Callable<T>) throws Exception` does?